sdk/apis/cache/v1alpha1/types_cachedresources_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2024 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestCachedResource(t *testing.T) {
+	testCases := []struct {
+		name     string
+		resource CachedResource
+	}{
+		{
+			name: "basic published resource",
+			resource: CachedResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "instances",
+					Namespace: "default",
+				},
+				Spec: CachedResourceSpec{
+					GroupVersionResource: GroupVersionResource{
+						Group:    "corp.com",
+						Resource: "instances",
+					},
+					LabelSelector: &metav1.LabelSelector{},
+				},
+				Status: CachedResourceStatus{
+					IdentityHash: "supersecrethash",
+				},
+			},
+		},
+		{
+			name: "minimal published resource",
+			resource: CachedResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "minimal",
+					Namespace: "default",
+				},
+				Spec: CachedResourceSpec{
+					GroupVersionResource: GroupVersionResource{
+						Group:    "corp.com",
+						Version:  "v1",
+						Resource: "minimal",
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Verify required fields are set
+			require.NotEmpty(t, tc.resource.Spec.Resource, "Resource field must not be empty")
```

Comment:
What does this test do? Isn't this something enforced by the schema definition, not necessarily by the Go type?

Reply:
Feels like this is leftover from one of the iterations. But don't remember what I wanted to test :D will delete.